RFC: Add cross references in subset related operator's docs.

AtsushiSakai opened the pull request:
I added cross references in subset related operator's docs to fix #22639.
I think adding cross references is a good idea to allows us to find the desired function easily.

AtsushiSakai committed:
Add cross references in subset related operator's docs.

bicycle1885 commented on base/abstractset.jl:
```diff
@@ -305,6 +306,7 @@ function ⊋ end
     ⊋(b, a) -> Bool
 
 Determines if `a` is a subset of, but not equal to, `b`.
+Here are some operators related to subset: [`issubset`](@ref)(⊆), [`⊈`](@ref).
```
```suggestion
Here are some operators related to subset: [`issubset`](@ref) (`⊆`), [`⊈`](@ref).
```

bicycle1885 commented on base/abstractset.jl:
```diff
@@ -328,6 +330,7 @@ function ⊉ end
     ⊉(b, a) -> Bool
 
 Negation of `⊆` and `⊇`, i.e. checks that `a` is not a subset of `b`.
+Here are some operators related to subset: [`issubset`](@ref)(⊆), [`⊊`](@ref).
```
```suggestion
Here are some operators related to subset: [`issubset`](@ref) (`⊆`), [`⊊`](@ref).
```

bicycle1885 reviewed (changes requested):
I think it would be better to quote operators in docstring.

mcabbott commented on base/abstractset.jl:
```diff
@@ -245,6 +245,7 @@ function ⊇ end
     ⊇(b, a) -> Bool
 
 Determine whether every element of `a` is also in `b`, using [`in`](@ref).
+Here are some operators related to subset: [`⊊`](@ref), [`⊈`](@ref).
```
I think the phrasing in many other docstrings is this, possibly with more explanation inserted:
```suggestion

See also [`⊊`](@ref), [`⊈`](@ref).
```

AtsushiSakai committed:
Update doc based on review

Co-authored-by: Kenta Sato <bicycle1885@gmail.com>

AtsushiSakai committed:
Update doc based on review

Co-authored-by: Kenta Sato <bicycle1885@gmail.com>

AtsushiSakai committed:
Update doc based on review

Co-authored-by: Michael Abbott <32575566+mcabbott@users.noreply.github.com>

AtsushiSakai committed:
Update doc based on review

AtsushiSakai commented:
I addressed all review comments. The CI failures looks like unrelated this PR change:
```
Error in testset cmdlineargs:
Error During Test at /buildworker/worker/tester_linux32/build/share/julia/test/cmdlineargs.jl:209
  Test threw exception
  Expression: read(`$exename -t $cpu_threads -e $code`, String) == string(cpu_threads)
  failed process: Process(`/buildworker/worker/tester_linux32/build/bin/julia -Cnative -J/buildworker/worker/tester_linux32/build/lib/julia/sys.so --depwarn=error --check-bounds=yes -g1 --startup-file=no --startup-file=no --color=no -t 256 -e 'print(Threads.nthreads())'`, ProcessSignaled(11)) [0]
```
